components/sidebar/sidebar_category/sidebar_category_menu/sidebar_category_menu.test.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {CategoryTypes} from 'mattermost-redux/constants/channel_categories';
+
+import {shallowWithIntl} from 'tests/helpers/intl-test-helper';
+
+import SidebarCategoryMenu from './sidebar_category_menu';
+
+describe('components/sidebar/sidebar_category/sidebar_category_menu', () => {
+    const baseProps = {
+        category: {
+            id: 'category1',
+            team_id: 'team1',
+            type: CategoryTypes.CUSTOM,
+            display_name: 'custom_category_1',
+        },
+        canCreatePublicChannel: true,
+        canCreatePrivateChannel: true,
+        isMuted: false,
+        actions: {},
+    };
+
+    test('should match snapshot', () => {
```

Comment:
Is there something specific we can look for in these tests like the existence of some component or how the props are passed into children? Just confirming that they match some snapshot doesn't do much since, if they break in the future, someone will likely just run the tests with `--update` without reading what actually changed.

Reply:
Most of the behaviour wrt. what's rendered into the snapshot is covered by the `baseProps`, so I think this test is still valuable. I don't think we need to also check for component existence as well, that just does what `toMatchSnapshot` already does.

We need to just be dilligent with PRs and make sure people aren't updating snapshots without checking if they changed something major in the component.

Reply:
It's not likely anyone touching this component won't have some understanding of what it does. It's not a global component being re-used by other parts of the app.

If a snapshot is going to break on this component, it'll only be from editing the component itself. I think whoever would see this snapshot will have enough info to know if they've broken something, it's pretty clear if the snapshot no longer renders the menu or one of the menu items.

If people are rubber-stamping without looking at any of the snapshot changes, that's a failure of the PR review system.

I can add some comments here or something to make sure other developers know what to look for if that would ease your mind, but I think we're being overzealous here on something that's just a verification step. Other tests will also be covering a lot of this behaviour, so I think we have enough safe-guards here.

Reply:
We're not the only ones that look at and modify this code, especially since we're open source, but even if we were, we're not going to remember how many of the details of this work in a few months. Clear tests like "when a user does X, Y happens" help to document how this is expected to work and ensures that the important parts of a component continue to work. Snapshot tests aren't very helpful since they fail as soon as anything changes, and when that happens, they don't do a good job saying what did change without combing through a hundred or more lines of snapshot.

Reply:
I can remove the snapshot test if you'd like. There's not much else really to test other than setting different `props` to the component to see which menu items show up. I think the snapshot is useful for that.

Reply:
I've added some additional checks to the snapshot tests.